IdentityCore/src/webview/response/MSIDWebOAuth2Response.m

```diff
@@ -26,8 +26,127 @@
 //------------------------------------------------------------------------------
 
 #import "MSIDWebOAuth2Response.h"
+#import "MSIDWebAADAuthResponse.h"
+#import "MSIDWebWPJAuthResponse.h"
 
 @implementation MSIDWebOAuth2Response
 
++ (MSIDWebOAuth2Response *)responseWithURL:(NSURL *)url
+                              requestState:(NSString *)requestState
+                             stateVerifier:(MSIDWebUIStateVerifier)stateVerifier
+                                   context:(id<MSIDRequestContext>)context
+                                     error:(NSError **)error;
+
+{
+    // This error case *really* shouldn't occur. If we're seeing it it's almost certainly a developer bug
+    if ([NSString msidIsStringNilOrBlank:url.absoluteString])
+    {
+        if (error){
+            *error = MSIDCreateError(MSIDOAuthErrorDomain, MSIDErrorNoAuthorizationResponse, @"No authorization response received from server.", nil, nil, nil, context.correlationId, nil);
+            return nil;
+        }
+    }
+    
+    // Check for WPJ response
+    if([url.absoluteString hasPrefix:@"msauth://"])
+    {
+        NSString *query = [url query];
+        NSDictionary *queryParams = [NSDictionary msidURLFormDecode:query];
+        NSString *appURLString = [queryParams objectForKey:@"app_link"];
+        
+        MSIDWebWPJAuthResponse *response = [MSIDWebWPJAuthResponse new];
```

Comment:
Let's discuss, if MSIDWebOAuth2Response should create its subclass.

Reply:
Like above, I'd like to see it moved.